### DISCUSSION TITLE
Build Error - anaconda3 OSX + crash report

### DISCUSSION THREAD

**Discussion by marmolRBG** [idaholab/moose#17722] [Q&A Getting Started] [answered]
Hello devs and community,

I have a major build failure when going to run the tests for the MOOSE build in my conda moose env. This started the other while working with Github and doing a make clean then and attempting to make my MOOSE app which was previously working. I ran an update with Big Sur, then after updating conda, and recompiling moose, the failure occurs at:

```
cd ~/projects/moose/test
make -j 4 # <--- build error
./run_tests -j 4 
```

I then went to the conda troubleshooting and failed to run the script:

```
cd /tmp
cat << EOF > hello.C
# include ...
```

The failure also  prevented me from running make altogether for other cpp applications. I could not isolate the problem to Xcode, anaconda etc, so I did a factory reset and reinstalled Big Sur 11.3, dev tools etc and attempt to reinstall MOOSE.

I then received the same failure messages after repeating the above steps. As per the MOOSE error output, I have attached the error and crash report here below.

Thank you very much in advance for any guidance,

Paul

[makeError.txt](https://github.com/idaholab/moose/files/6410613/makeError.txt)

[mooseEnvBuildTest.txt](https://github.com/idaholab/moose/files/6410612/mooseEnvBuildTest.txt)

[pcrecpp-7ca045.cpp.txt](https://github.com/idaholab/moose/files/6410614/pcrecpp-7ca045.cpp.txt)

[pcrecpp-7ca045.sh.txt](https://github.com/idaholab/moose/files/6410615/pcrecpp-7ca045.sh.txt)

**Comment by milljm** [comment]
This may be a _new_ Xcode issue. I know a recent update occurred, but I have yet to test it. I'll do that now and report back.

**Reply by milljm** [reply]
I get the same error when running through the normal installation routine on latest BigSur (11.3). Prior to the update, things were working (aside from Peacock). The update included a bump to Xcode, and a large system update (11.2.3 --> 11.3).

Most likely, the fix will involve downgrading Xcode. I'll begin there, and report back.

**Reply by milljm** [reply]
Downgrading Xcode to 12.4 solved my issue. But not without some strangeness beforehand.

I already had Xcode 12.4 installed (according to Xcode, About). Seemingly there exists an update between this and 12.5?

I began troubleshooting by installing 12.5. Received the same error as before. I then downgraded to 11.4, same error. Installed 12.4, error went away.

During the mix of all this I had also tried installing Xcode CLT 11.4, 11.5, and 12.5. Which had no effect. Admittedly `xcode-select -p` continued to point to my installation of Xcode, so I doubt my system ever truly used my CLT installation(s).

The end result is that I removed Xcode CLT entirely:`sudo rm -rf /Library/Developer/CommandLineTools`, and installed Xcode 12.4.

**Reply by pbalest** [reply]
same here:

`In file included from /Users/balep/projects/moose/framework/contrib/pcre/src/pcrecpp.cc:36:
/Users/balep/opt/anaconda3/envs/moose00/bin/../include/c++/v1/stdlib.h:142:34: error: unknown type name 'ldiv_t'
inline _LIBCPP_INLINE_VISIBILITY ldiv_t div(long __x, long __y) _NOEXCEPT {`

I'll try downgrading

**Reply by pbalest** [reply]
@milljm how you install previous version of Xcode and Xcode CLT?

you remove the new one `sudo rm -rf /Library/Developer/CommandLineTools` (and the app) download the previous version from apple install it and then point at it with `sudo xcode-select -s /Applications/XcodeVersion.app/` ?

**Reply by marmolRBG** [reply]
Hello Jason,

Thanks for investigating and confirming this. Strangeness for me as well, as the failure was not linked to the Big Sur update, and I had an older (<=v12.4) version of Xcode that was working previously. Sorry I can't provide more explicit troubleshooting results since I reset the system.

Unfortunately it seems like that command to remove the command line tools will also mess up any other code/toolkit requiring to be built with Xcode, is that correct?

I will also try the above steps  Paolo and you  mentioned.

**Reply by milljm** [reply]
Believe it or not, that is Apple's suggested means of removing Xcode CLT (Command Line Tools):

https://developer.apple.com/library/archive/technotes/tn2339/_index.html#//apple_ref/doc/uid/DTS40014588-CH1-HOW_CAN_I_UNINSTALL_THE_COMMAND_LINE_TOOLS_

which translates to:
```bash
sudo rm -rf /Library/Developer/CommandLineTools
```

Installing any version of Xcode is a simple double-click on the Xcode.xip compressed file (Xcode.app is extracted), drag Xcode.app to /Applications, which will replace your current Xcode. After that, you run /Applications/Xcode which allows it to finalize anything it may need to do, and you are done.

You may need to adjust your xcode-select path, if you were strictly using CLT only:

```bash
sudo xcode-select --switch /Applications/Xcode.app
```

Also, the reset feature might work best:
```bash
sudo xcode-select --reset
```

@marmolRBG Good to know, that you too had 12.4 to begin with. Very odd indeed, as that is the version I _ended up with_, which is what I began with... but now it works. Maybe Xcode needed to be... um, reinstalled?

**Reply by pbalest** [reply]
take soooooooooooooooooooo long to downlad Xcode😆... I'll let you know if it works Thanks @milljm and @GiudGiud  to point me here.

**Reply by pbalest** [reply]
It worked! Thanks again!

**Reply by nolmacdonald** [reply]
Can confirm I had the same issue (XCode 12.5 on OS X) and downgraded to 12.4 w/ success.

**Comment by cticenhour** [comment, marked as answer]
For those who experience a similar issue in the future, I am re-answering this question now that MOOSE also supports Xcode 12.5, 13, and 13.1. Please update your conda packages to the newest versions by performing 
```
conda update --all
```
within your MOOSE conda environment. See #19289 for more information.
